json-endpoints/class.wpcom-json-api-export-start-endpoint.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+// POST /sites/%s/exports/start
+class WPCOM_JSON_API_Export_Start_Endpoint extends WPCOM_JSON_API_Endpoint {
+	protected $needed_capabilities = 'export';
+
+	function callback( $path = '', $blog_id = 0 ) {
+		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
+		if ( is_wp_error( $blog_id ) ) {
+			return $blog_id;
+		}
+
+		if ( !current_user_can( $this->needed_capabilities ) ) {
+			return new WP_Error( 'unauthorized', 'User cannot export', 403 );
+		}
+		$args = $this->input();
+
+		// Set up args array
+		$args = $this->setup_args( $args );
+
+		/**
+		 * Filters the export args.
+		 *
+		 * @since 3.5.0
+		 *
+		 * @param array $args The arguments to send to the exporter.
+		 */
+		$args = apply_filters( 'export_args', $args );
+
+		// Using output buffering to catch the exporter response and write it to a file.
+		include( ABSPATH . 'wp-admin/includes/export.php' );
+		ob_start();
+		export_wp( $args);
+		$string_data = ob_get_clean();
+
+		// Export file is saved to the uploads folder.
+		$file_name = '/export_' . wp_create_nonce() . '.xml';
+		$file_path = wp_upload_dir()['path'] . $file_name;
```

Comment:
This will not work in PHP versions 5.3.x and earlier, which are [supported by WordPress](https://wordpress.org/about/requirements/) and therefore Jetpack:

>As of PHP 5.4 it is possible to array dereference the result of a function or method call directly. Before it was only possible using a temporary variable.

https://secure.php.net/manual/en/language.types.array.php#example-62